dask_expr/_expr.py

```diff
@@ -1275,6 +1276,196 @@ def _task(self, index: int):
             )
 
 
+class MapOverlap(MapPartitions):
+    _parameters = [
+        "frame",
+        "func",
+        "before",
+        "after",
+        "meta",
+        "enforce_metadata",
+        "transform_divisions",
+        "clear_divisions",
+        "kwargs",
+    ]
+    _defaults = {
+        "meta": None,
+        "enfore_metadata": True,
+        "transform_divisions": True,
+        "kwargs": None,
+        "clear_divisions": False,
+    }
+
+    @functools.cached_property
+    def _kwargs(self) -> dict:
+        kwargs = self.kwargs
+        if kwargs is None:
+            kwargs = {}
+        kwargs = kwargs.copy()
+        kwargs.update({"before": self.before, "after": self.after, "func": self.func})
+        return kwargs
+
+    @functools.cached_property
+    def _meta(self):
+        meta = self.operand("meta")
+        args = [arg._meta if isinstance(arg, Expr) else arg for arg in self.args]
+        return _get_meta_map_partitions(args, [], self.func, self.kwargs, meta, None)
+
+    @functools.cached_property
+    def before(self):
+        before = self.operand("before")
+        if isinstance(before, str):
+            return pd.to_timedelta(before)
+        elif isinstance(before, numbers.Integral):
+            if before < 0:
+                raise ValueError("before must be positive integer")
+        return before
+
+    @functools.cached_property
+    def after(self):
+        after = self.operand("after")
+        if isinstance(after, str):
+            return pd.to_timedelta(after)
+        elif isinstance(after, numbers.Integral):
+            if after < 0:
+                raise ValueError("after must be positive integer")
+        return after
+
+    def _lower(self):
+        overlapped = MapOverlapInterleavePartitions(self.frame, self.before, self.after)
+
+        return MapPartitions(
+            overlapped,
+            _overlap_chunk,
+            self._meta,
+            self.enforce_metadata,
+            self.transform_divisions,
+            self.clear_divisions,
+            self._kwargs,
+        )
+
+
+class MapOverlapInterleavePartitions(Expr):
```

Comment:
Another note to self: The `Interleave` term comes from the fact that we are extracting before/after rows for each partition and effectively "interleaving" the duplicated data between the "real" partitions. I don't love this name, but I also don't really have a better suggestion (maybe `CreateOverlappingPartitions`?)

Reply:
No strong opinion either way